lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/stateful/__tests__/NxStatefulCollapsibleMultiSelect.test.tsx

```diff
@@ -4,93 +4,543 @@
+      ...props,
+      filterThreshold: 1
+    });
+
+    it('renders an input with type="text" iff when the number of options ' +
+      'is greater than filterThreshold', function() {
+      expect(quickRender().queryByRole('textbox')).not.toBeInTheDocument();
+      expect(filterView().getByRole('textbox')).toHaveAttribute('type', 'text');
+    });
+
+    it('sets the filter input placeholder as specified by filterPlaceholder, ' +
+      'default is "filter"', function() {
+      const viewWithPlaceholder = filterView({ filterPlaceholder: 'filter cats' });
+
+      expect(filterView().getByRole('textbox')).toHaveAttribute('placeholder', 'filter');
+      expect(viewWithPlaceholder.getByRole('textbox')).toHaveAttribute('placeholder', 'filter cats');
+    });
+
+    it('renders a button with an accessible name of "Clear filter"', async function() {
+      const view = filterView();
+
+      await runTimers();
+      const clearBtn = view.getByRole('button', { name: 'Clear filter' });
+
+      expect(clearBtn).toBeInTheDocument();
+      expect(clearBtn).toHaveAccessibleName('Clear filter');
+    });
+
+    it('clears the input when the clear button is clicked', async function() {
+      const user = userEvent.setup(),
+          view = filterView(),
+          inputEl = view.getByRole('textbox');
+
+      await runTimers();
+
+      const clearBtn = view.getByRole('button', { name: 'Clear filter' });
+
+      await user.click(clearBtn);
+      expect(inputEl).toHaveValue('');
+    });
+
+    it('clears the input when Escape key is pressed', async function() {
+      const user = userEvent.setup(),
+          view = filterView(),
+          inputEl = view.getByRole('textbox');
+
+      await user.type(inputEl, 'f');
+
+      expect(inputEl).toHaveValue('f');
+
+      await user.keyboard('[Escape]');
+
+      expect(inputEl).toHaveValue('');
+    });
+
+    it('calls preventDefault on the event when Escape key is pressed and the value is not empty', async function() {
+      const user = userEvent.setup(),
+          { getByRole } = filterView(),
+          inputEl = getByRole('textbox'),
+          keyEvent = createEvent.keyDown(inputEl, { key: 'Escape' });
+
+      await user.type(inputEl, 'f');
+      expect(inputEl).toHaveValue('f');
+
+      fireEvent(inputEl, keyEvent);
+
+      expect(keyEvent.defaultPrevented).toBe(true);
+    });
+
+    it('does not call preventDefault on the event when Escape key is pressed and the value is empty', () => {
+      const { getByRole } = filterView(),
+          inputEl = getByRole('textbox'),
+          keyEvent = createEvent.keyDown(inputEl, { key: 'Escape' });
+
+      expect(inputEl).toHaveValue('');
+
+      inputEl.focus();
+      fireEvent(inputEl, keyEvent);
+
+      expect(keyEvent.defaultPrevented).toBe(false);
     });
-    expect(shallowRender).toHaveProp('isOpen', true);
-    shallowRender.simulate('toggleCollapse');
-    expect(shallowRender).toHaveProp('isOpen', false);
   });
```

Comment:
The stateful version of this component handles the filtering internally.  That needs to be tested.

Reply:
Updated here f0e6a20b5cc00424fde36d4b1d535f46452f155b

Reply:
> Updated here https://github.com/sonatype/sonatype-react-shared-components/commit/f0e6a20b5cc00424fde36d4b1d535f46452f155b

I think that needs to be fleshed out more.  For one thing, when the filter is active, we should check not just the number of options still present but which ones they are.  Additionally, more filtering cases should be checked.  For instance case-sensitivity, filtering by strings that match other parts besides the beginning of the word, and also something that tests the `fuzzyFilter` functionality of `NxStatefulCollapsibleMultiSelect` to at least some extent.

Reply:
Added more tests here 378b2fb2a90fe03deb2152edc9139c47c7c56a36